Source/CBLDatabase+Internal.m

```diff
@@ -166,16 +176,25 @@ - (BOOL) open: (NSError**)outError {
         secondaryStorage = NSClassFromString(@"CBL_SQLiteStorage");
     // Use primary unless dir already contains a db created by secondary:
     id<CBL_Storage> storage = [[secondaryStorage alloc] init];
-    if (![storage databaseExistsIn: _dir])
+    if (![storage databaseExistsAtPath: _path])
         storage = [[primaryStorage alloc] init];
-    LogTo(CBLDatabase, @"Using %@ for db at %@", [storage class], _dir);
+
+    if ([storage isKindOfClass: NSClassFromString(@"CBL_ForestDBStorage")]) {
```

Comment:
Don't hardcode decisions based on storage type! That belongs in the storage implementation. Put the -createDirectoryAtPath call in the storage's openAtPath: method.